catroid/src/org/catrobat/catroid/physics/PhysicsObject.java

```diff
@@ -123,12 +123,10 @@ private void calculateCircumference() {
 			return;
 		}
 
-		calculateAABB();
-		float max1 = Math.max(getMassCenter().dst(fixtureAABBUpperRight), getMassCenter().dst(fixtureAABBLowerLeft));
-		Vector2 aabbLowerRight = new Vector2(fixtureAABBUpperRight.x, fixtureAABBLowerLeft.y);
-		Vector2 aaabbUpperLeft = new Vector2(fixtureAABBLowerLeft.x, fixtureAABBUpperRight.y);
-		float max2 = Math.max(getMassCenter().dst(aabbLowerRight), getMassCenter().dst(aaabbUpperLeft));
-		circumference = Math.max(max1, max2);
+		Vector2 dimensions = getBoundaryBoxDimensions();
+		float a = dimensions.x / 2;
+		float b = dimensions.y / 2;
+		circumference = PhysicsWorldConverter.convertNormalToBox2dCoordinate((float) Math.sqrt(a * a + b * b));
```

Comment:
Vector2 provides a function for the euclidean distance, ```len```
Variable names a and b... no no :smile_cat:. If this should be the center, the name center should be in the variable name.